src/utils/server-utils.ts

```diff
@@ -0,0 +1,165 @@
+import moment from 'moment';
+import { xml2js } from 'xml-js';
+import { Map } from 'immutable';
+import { merge, unset, get, isString, union, isEmpty } from 'lodash';
+import { format, parse } from 'url';
+
+import config from '../config/prism.json';
+import { AvailableDates } from '../config/types';
+
+const xml2jsOptions = {
+  compact: true,
+  trim: true,
+  ignoreComment: true,
+};
+
+export function formatServerUri(
+  serverUri: string,
+  queryProp: { [key: string]: string | boolean | number },
+  selectedDate?: number,
+) {
+  // The second arg of 'parse' allows us to have 'query' as an object
+  const { query, ...parsedUrl } = parse(serverUri, true);
+
+  // Removing 'search' to be able to format by 'query'
+  unset(parsedUrl, 'search');
+
+  const selectedDateString = selectedDate
+    ? moment(selectedDate).format('YYYY-MM-DD')
+    : '';
+
+  return decodeURI(
+    format({
+      ...parsedUrl,
+      query: merge(query, queryProp, { time: selectedDateString }),
```

Comment:
`{ time: selectedDateString }` should probably be conditionally included depending on whether we have a `selectedDate`?

Also, you can probably just use normal es6 object: `{...query, ...queryProp, ...selectedDate && { time: selectedDateString }}`